python/pyiceberg/table/base.py

```diff
@@ -15,17 +15,17 @@
 #  specific language governing permissions and limitations
 #  under the License.
 
-from __future__ import annotations
+from typing import Dict, Optional, Union
 
-from abc import ABC
+from pydantic import Field
 
 from pyiceberg.catalog.base import Identifier
+from pyiceberg.table.metadata import TableMetadataV1, TableMetadataV2
+from pyiceberg.utils.iceberg_base_model import IcebergBaseModel
 
 
-class Table(ABC):
-    """Placeholder for Table managed by the Catalog that points to the current Table Metadata.
-
-    To be implemented by https://github.com/apache/iceberg/issues/3227
-    """
-
-    identifier: str | Identifier
+class Table(IcebergBaseModel):
```

Comment:
Does `Table` need to be serializable? I'm curious why Table extends `IcebergBaseModel`. It seems like it should be a regular class to me.

Reply:
Yes, it needs to be serializable, and it is the equivalent of the response of the REST API: https://github.com/apache/iceberg/blob/master/open-api/rest-catalog-open-api.yaml#L1481-L1503

The `CreateTableResponse` is an alias of the `LoadTableResponse`.

Reply:
I don't think the `LoadTableResponse` should be the same class as Table. Table is part of the public API and will expose a lot of methods and operations to users. `LoadTableResponse` is specific to the REST catalog and includes fields that aren't part of the table (`config`).

Reply:
This is defined by the Catalog: https://github.com/apache/iceberg/blob/84951419a9ab0aa39f9304b3dce91a179f4e643b/python/pyiceberg/catalog/base.py#L54-L77

How about subclassing the table:
```
class RestTable(Table):
    config: Dict[str, str] = Field(default_factory=dict)
```
And moving the specific items to there.

Reply:
I think we will want to evolve this in the future, but it is okay for now.

We will need to think about whether we want to use the same `TableOperations` abstraction that Java uses. That allows us to have a base table implementation that can work with any catalog. In that design, the table wraps a `TableMetadata` that is handled (refreshed, committed) by `TableOperations`.